crates/compiler/can/src/def.rs

```diff
@@ -2880,6 +2941,188 @@ fn to_pending_value_def<'a>(
+                module: module_name.clone(),
+            }
+            .into_pq_module_name(env.opt_shorthand);
+
+            let module_id = env
+                .qualified_module_ids
+                .get_id(&pq_module_name)
+                .expect("Module id should have been added in load");
+
+            let name_with_alias = match module_import.alias {
+                Some(alias) => ModuleName::from(alias.item.value.as_str()),
+                None => module_name.clone(),
+            };
+
+            if let Err(existing_import) =
+                scope
+                    .modules
+                    .insert(name_with_alias.clone(), module_id, region)
+            {
+                env.problems.push(Problem::ImportNameConflict {
+                    name: name_with_alias,
+                    is_alias: module_import.alias.is_some(),
+                    new_module_id: module_id,
+                    new_import_region: region,
+                    existing_import,
+                });
+
+                return PendingValue::ImportNameConflict;
+            }
+
+            let mut exposed_symbols;
+
+            let is_automatically_imported =
+                !env.home.is_builtin() && module_id.is_automatically_imported();
+
+            match module_import.exposed {
+                None => {
+                    exposed_symbols = Vec::new();
+
+                    if is_automatically_imported {
+                        env.problems
+                            .push(Problem::ExplicitBuiltinImport(module_id, region));
+                    }
+                }
+                Some(exposed_kw) => {
+                    let exposed_ids = env
+                        .dep_idents
+                        .get(&module_id)
+                        .expect("Module id should have been added in load");
+
+                    exposed_symbols = Vec::with_capacity(exposed_kw.item.len());
+
+                    for loc_name in exposed_kw.item.items {
+                        let exposed_name = loc_name.value.item();
+                        let name = exposed_name.as_str();
+                        let ident = Ident::from(name);
+
+                        match exposed_ids.get_id(name) {
+                            Some(ident_id) => {
+                                let symbol = Symbol::new(module_id, ident_id);
+                                exposed_symbols.push((symbol, loc_name.region));
+
+                                match scope.import_symbol(ident, symbol, loc_name.region) {
+                                    Ok(()) => {}
+                                    Err((_shadowed_symbol, existing_symbol_region)) => {
+                                        if is_automatically_imported
+                                            && Symbol::builtin_types_in_scope(module_id)
+                                                .iter()
+                                                .any(|(_, (s, _))| *s == symbol)
```

Comment:
Can this be made a method on `Symbol`? e.g. `Symbol::is_builtin_symbol_in_scope`